### PR TITLE
[OSX] Check dylib and binaries for relative paths

### PR DESCRIPTION
**Pull request by sjoelund**

### CODE REVIEW

**Comment by AnHeuermann** [comment]
We need also a sanity check for build/bin/OMCppOSUSimulation for Mac. `LC_RPATH` was wrong but until now not cough and will not be found by `otool -L`.

Doing a simple `./build/bin/OMCppOSUSimulation --help` and check if it succeeded could be enough.

Also I can't see what's missing now for OMCppOSUSimulation. All dylibs are at `/usr`, `/opt` or `@rpath`.

```bash
omodmac:OpenModelica andreas$ ( find build/bin -type f -exec otool -L {} ";" | tr -d "" | grep -v : | grep -v "^[/@]" )
        @rpath/libOpenModelicaCompiler.dylib (compatibility version 0.0.0, current version 0.0.0)
        /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1226.10.1)
        @rpath/libOMCppExtensionUtilities.dylib (compatibility version 0.0.0, current version 0.0.0)
        @rpath/libOMCppModelicaUtilities.dylib (compatibility version 0.0.0, current version 0.0.0)
        /opt/openmodelica/lib/libboost_filesystem-mt.dylib (compatibility version 0.0.0, current version 0.0.0)
        /opt/openmodelica/lib/libboost_system-mt.dylib (compatibility version 0.0.0, current version 0.0.0)
        /opt/openmodelica/lib/libboost_serialization-mt.dylib (compatibility version 0.0.0, current version 0.0.0)
        /opt/openmodelica/lib/libboost_program_options-mt.dylib (compatibility version 0.0.0, current version 0.0.0)
        /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1226.10.1)
        /usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 120.1.0)
```

**Comment by sjoelund** [comment]
I added that check + OMEdit

**Comment by AnHeuermann** [comment]
So `./build/bin/OMCppOSUSimulation` will not return 0 when running with `--help`. Not very helpful.
Just remove the check for it and fix that in a different PR?

File: Jenkinsfile [hunk 1]
```diff
@@ -102,6 +102,10 @@ pipeline {
                 sh "${env.GMAKE} --version"
                 common.buildOMC('cc', 'c++', "OMPCC='gcc-mp-5 -fopenmp -mno-avx' GNUCXX=g++-mp-5 FC=gfortran-mp-5 LDFLAGS=-L${env.MACPORTS}/lib CPPFLAGS=-I${env.MACPORTS}/include --without-omlibrary")
                 common.buildGUI('', false)
+                sh name: "Look for relative paths in dylibs", script: '! ( find build/ -name "*.dylib" -exec otool -L {} ";" | tr -d "\t" | grep -v : | grep -v "^[/@]" )'
+                sh name: "Look for relative paths in bin folder", script: '! ( find build/bin -type f -exec otool -L {} ";" | tr -d "\t" | grep -v : | grep -v "^[/@]" )'
+                sh name: "Sanity check for OMSI", script: "./build/bin/OMCppOSUSimulation --help"
```
Comment (AnHeuermann) [inline Jenkinsfile:107]: It's the sanity check for the Cpp runtime, not really for OMSI.

**Comment by sjoelund** [comment]
Might as well fix that so it returns 0? It's a good check to have

**Comment by AnHeuermann** [comment]
Could it be that OMEdit is not build for this target?

**Comment by adrpo** [comment]
Is not in the usual place on Mac (of course, why should it be easy?!?)
Is called OMEdit.app and is in build/Applications/

```
if [ ".app" = ".app" ]; then mkdir -p /Users/hudson/jenkins-slave/ws/OpenModelica_PR-916/build/Applications/ ; cp -rp bin/OMEdit.app /Users/hudson/jenkins-slave/ws/OpenModelica_PR-916/build/Applications/ ; else cp -p bin/OMEdit /Users/hudson/jenkins-slave/ws/OpenModelica_PR-916/build/bin/; fi

cp -p bin/libOMEdit.a /Users/hudson/jenkins-slave/ws/OpenModelica_PR-916/build/lib/x86_64-darwin15.4.0/omc/
```

**Comment by sjoelund** [comment]
True. OMEdit hides it inside the .app folder. I can't remember the full name right now

**Pull request merged**